components/PlantList.tsx

```diff
@@ -2,21 +2,79 @@ import React, { useEffect, useState } from 'react';
 import { getPlantSeasonality } from '@/api/supabase/queries/plantSeasonality';
 import { Plant } from '@/types/schema';
 
-export const PlantList = () => {
+interface PlantListProps {
+  harvest_season: string;
+  planting_type: string;
+  growing_season: string;
+}
+
+export const PlantList = (props: PlantListProps) => {
   const [plants, setPlants] = useState<Plant[]>([]);
+  const growingSeasonToMonth = new Map<string, string[]>([
+    ['Spring', ['MARCH', 'APRIL', 'MAY']],
+    ['Summer', ['JUNE', 'JULY', 'AUGUST']],
+    ['Fall', ['SEPTEMBER', 'OCTOBER', 'NOVEMBER']],
+    ['Winter', ['DECEMBER', 'JANUARY', 'FEBRUARY']],
+  ]);
 
   useEffect(() => {
     const fetchPlantSeasonality = async () => {
-      const plantList = await getPlantSeasonality('Tennessee');
+      // gets plants in Tennessee by default
+      const plantList = await getPlantSeasonality('TENNESSEE');
       setPlants(plantList);
     };
 
     fetchPlantSeasonality();
   }, []);
 
+  const checkGrowingSeason = (plant: Plant) => {
+    if (!props.growing_season) {
+      return true;
+    }
+
+    const months = growingSeasonToMonth.get(props.growing_season);
+
+    return (
+      months?.includes(plant.indoors_start) ||
+      months?.includes(plant.indoors_end) ||
+      months?.includes(plant.outdoors_start) ||
+      months?.includes(plant.outdoors_end)
+    );
+  };
+
+  const checkHarvestSeason = (plant: Plant) => {
+    if (!props.harvest_season) {
+      return true;
+    }
+
+    return plant.harvest_season === props.harvest_season.toLocaleUpperCase();
+  };
+
+  const checkPlantingType = (plant: Plant) => {
+    if (!props.planting_type) {
+      return true;
+    }
+
+    if (props.planting_type === 'Start Seeds Indoors') {
+      return plant.indoors_start !== null && plant.indoors_start !== null;
+    } else if (props.planting_type === 'Start Seeds Outdoors') {
+      return plant.outdoors_start !== null && plant.outdoors_start !== null;
+    } else if (props.planting_type === 'Plant Seedlings/Transplant Outdoors') {
+      return plant.transplant_start !== null && plant.transplant_end !== null;
+    }
+  };
+
+  const filterPlantList = (plant: Plant) => {
+    return (
+      checkGrowingSeason(plant) &&
```

Comment:
i rlly like this structure! it's super clean; great job! 
quick note: in the next sprint we'll add a state filter too